File: production/client/src/main/scala/kpn/client/components/common/FactInfo.scala
// TODO migrate to Angular
package kpn.client.components.common

import kpn.shared.Fact
import kpn.shared.common.Ref

case class FactInfo(
  fact: Fact,
  networkRef: Option[Ref] = None,
  routeRef: Option[Ref] = None,
  nodeRef: Option[Ref] = None
)

File: client/src/app/analysis/fact/fact-info.ts
import { Fact } from '@api/custom/fact';
import { Ref } from '@api/common/common/ref';

export class FactInfo {
  constructor(
    public fact: Fact,
    public networkRef?: Ref,
    public routeRef?: Ref,
    public nodeRef?: Ref,
    public unexpectedNodeIds?: number[],
    public unexpectedRelationIds?: number[]
  ) {}
}
